hi-ml/src/health_ml/run_ml.py

```diff
@@ -274,81 +268,78 @@ def is_crossval_disabled_or_child_0(self) -> bool:
             return self.container.crossval_index == 0
         return True
 
-    def get_trainer_for_inference(self, checkpoint_path: Optional[Path] = None) -> Trainer:
-        # We run inference on a single device because distributed strategies such as DDP use DistributedSampler
-        # internally, which replicates some samples to make sure all devices have the same batch size in case of
-        # uneven inputs.
+    def set_trainer_for_inference(self) -> None:
+        """ Set the runner's trainer for inference: validation or test.
+        We run inference on a single device because distributed strategies such as DDP use DistributedSampler
+        internally, which replicates some samples to make sure all devices have the same batch size in case of
+        uneven inputs which biases the results."""
         mlflow_run_id = get_mlflow_run_id_from_previous_loggers(self.trainer)
         self.container.max_num_gpus = 1
-
-        if self.container.run_inference_only:
-            assert checkpoint_path is not None
-        else:
-            self.validate_model_weights()
-
-        trainer, _ = create_lightning_trainer(
+        self.trainer, _ = create_lightning_trainer(
             container=self.container,
-            resume_from_checkpoint=checkpoint_path,
             num_nodes=1,
             azureml_run_for_logging=self.azureml_run_for_logging,
             mlflow_run_for_logging=mlflow_run_id
         )
-        return trainer
+
+    def init_inference(self) -> None:
```

Comment:
Should the title of this method reflect that it is the required initialisation for validation, as well as inference?

Reply:
I think that we are misusing the term inference. What we are actually doing is an evaluation on the test and validation set. Inference is done with trainer.predict when we don't have a ground truth at all 
<img width="426" alt="image" src="https://user-images.githubusercontent.com/37396332/212883754-43e555d6-0eaa-4506-9947-b4e0704c5830.png">
Whereas test and validate 
<img width="442" alt="image" src="https://user-images.githubusercontent.com/37396332/212883938-3659a4c5-198a-4e80-b88e-16d0b209aadc.png">
<img width="458" alt="image" src="https://user-images.githubusercontent.com/37396332/212883961-16ba9feb-d85f-426d-8e4e-e70975bb03e1.png">
<img width="425" alt="image" src="https://user-images.githubusercontent.com/37396332/212883996-b75985b9-df29-414f-80f3-20352f0ce078.png">